src/naming/Naming_AST.ml

```diff
@@ -590,6 +590,7 @@ let resolve lang prog =
                     (* TODO handle nested destructuring? *)
                 | _ -> ())
               fields
+        | { name = EPattern (PatId (id, id_info)); _ }, VarDef { vinit; vtype }
```

Comment:
Could you add a comment here with a brief example so that future readers understand why we're handling this case? Also, do you think there should be a TODO to handle more pattern kinds?

Reply:
Done. Thanks for the comments!